test/throttle/core_test.clj

```diff
@@ -114,3 +114,66 @@
    (do (Thread/sleep 25)
        (attempt 1)
        (count @(:attempts test-throttler)))])
+
+
+;;; # tests for with-thottling
+
+;; Check that no attempts are recorded for empty throttled body
+(expect
+  0
+  (do
+    (reset! (:attempts test-throttler) '())
+    (throttle/with-throttling test-throttler :test)
+    (count @(:attempts test-throttler))))
+
+(let [threshold            (:attempts-threshold test-throttler)
```

Comment:
I think it's a little clearer if you break the tests out into separate top-level forms rather than maintaining and resetting state between each test. That way you can run one of the tests in isolation without having to run the ones it depends on for state.

Since Throttlers are just simple records it should be easy enough to start each test with a throttler in the state you want it in. You can move your helper fns `login-failed` etc. to the top-level as well and continue using them in multiple tests

Reply:
I would recommend reading thru the style guide and following everything in there, where there's code in Metabase that isn't following the style guide it's probably because it was written before the style guide had rules about it, so don't take it as an example of how to do stuff.